src/vmm/src/vmm_config/vsock.rs

```diff
@@ -16,20 +42,125 @@ pub struct VsockDeviceConfig {
     pub uds_path: String,
 }
 
-/// Errors associated with `VsockDeviceConfig`.
-#[derive(Debug)]
-pub enum VsockError {
-    /// The update is not allowed after booting the microvm.
-    UpdateNotAllowedPostBoot,
+struct VsockAndUnixPath {
+    vsock: Arc<Mutex<Vsock<VsockUnixBackend>>>,
+    uds_path: String,
 }
 
-impl Display for VsockError {
-    fn fmt(&self, f: &mut Formatter) -> Result {
-        use self::VsockError::*;
-        match *self {
-            UpdateNotAllowedPostBoot => {
-                write!(f, "The update operation is not allowed after boot.",)
+/// A builder of Vsock with Unix backend from 'VsockDeviceConfig'.
+#[derive(Default)]
+pub struct VsockStore {
+    inner: Option<VsockAndUnixPath>,
+}
+
+impl VsockStore {
+    /// Creates an empty Vsock with Unix backend Store.
+    pub fn new() -> Self {
+        Self { inner: None }
+    }
+
+    /// Inserts a Unix backend Vsock in the store.
+    /// If an entry already exists, it will overwrite it.
+    pub fn insert(&mut self, cfg: VsockDeviceConfig) -> Result<()> {
+        // Make sure to drop the old one and remove the socket before creating a new one.
+        if let Some(existing) = self.inner.take() {
+            let _ = std::fs::remove_file(existing.uds_path);
```

Comment:
We are not checking/returning the remove_file error. 
```suggestion
            std::fs::remove_file(existing.uds_path)?;
```

Reply:
This is attempting to remove any leftover uds. I thought we can ignore failures here and report errors in Vsock creation where the right error will be reported (permission errors for example).

Getting the right error here feels a bit like duplicating logic.

I will add a comment, or if you insist I will duplicate the error handling logic.

Reply:
I agree that the error should be propagated and the backend creation should fail fast. If there's a leftover that can't be removed, it could indicate that something went wrong before, and give the user a chance to diagnose, or maybe someone wants to keep using the previous uds, or some other strange case that might lead to trouble.

Reply:
done